devnotes/documentation.md

```diff
@@ -131,22 +143,44 @@ In order to build coreboot image, follow the steps below:
 3. Flash the binary by replacing HBB partition (execute from BMC):
 
    ```
-   pflash -e -P HBB -p /tmp/coreboot.rom
+   pflash -e -P HBB -p /tmp/coreboot.signed.ecc
    ```
+   > Again, if that file is not present, use `coreboot.rom` instead
 
    Answer yes to the prompt and wait for the process to finish.
 
-4. Log into the BMC GUI again at https://<IP>/. Enter the Server power
-   operations (https://<IP>/#/server-control/power-operations) and invoke warm
+4. Log into the BMC GUI again at https://\<BMC_IP\>/. Enter the Server power
+   operations (https://\<BMC_IP\>/#/server-control/power-operations) and invoke warm
    reboot. Then move to Serial over LAN remote console
-   (https://<IP>/#/server-control/remote-console)
+   (https://\<BMC_IP\>/#/server-control/remote-console)
 
    Wait for a while until coreboot shows up:
 
-   .center[.image-90[![](images/cb_bootblock.png)]]
+   ![](../images/cb_bootblock.png)
 
-5. Enjoy coreboot bootblock running on Talos II.
+5. Enjoy the coreboot running on Talos II.
 
-> OPTIONAL: in order to recovery the platform quickly to healthy state, flash
-> the HBB partition back with:
+> **Optional:** In order to recovery the platform quickly to healthy state, flash
+> the HBB partition back with:\
 > `pflash -e -P HBB -p /tmp/hbb.bin`
+
+## Running the coreboot on QEMU
+
+Please keep in mind, that QEMU doesn't implement many of the HW properties,
+that Talos II has. There may be many compatibility issues, or registers missing.
+
+1. Clone the QEMU repository
+   ```
+   git clone git@github.com:qemu/qemu.git
+   # or HTTPS alternatively
+   git clone https://github.com/qemu/qemu.git
+   ```
+2. Build the QEMU ppc64 version
+   ```
+   cd qemu
+    ./configure --target-list=ppc64-softmmu && make
+   ```
+3. Start QEMU with coreboot image
+   ````
+   ./qemu/build/qemu-system-ppc64 -M powernv,hb-mode=on --cpu power9 --bios 'coreboot/build/coreboot.rom' -d unimp,guest_errors -serial stdio
```

Comment:
Might be worth mentioning how to add PNOR:
```
-drive file=flash.pnor,format=raw,readonly=on,if=mtd
```

Reply:
Added the information